### PR TITLE
Improve creature numbers handling in castle interface bottom left part

### PR DESCRIPTION
**Pull request by dydzio0614**
Default label now matches H3, also improved support for HD Mod way (disabled by default) - for now accesible via "/set availableCreaturesAsLabel on"

### CODE REVIEW

**Comment by kambala-decapitator** [comment]
> for now accesible via "/set availableCreaturesAsLabel on"

I'd rather make it available in json config (and in the launcher)

**Comment by IvanSavenko** [comment]
> > for now accesible via "/set availableCreaturesAsLabel on"
> 
> I'd rather make it available in json config (and in the launcher)

Seconded.

Also, I propose to add new section in our settings json, let's call it guiTweaks (or something like that) and use it for all changes like this that deviate from H3

**Comment by dydzio0614** [comment]
I want to change this after finishing VCMI settings

File: client/windows/CCastleInterface.cpp [hunk 1]
```diff
@@ -1042,10 +1057,12 @@ void CCreaInfo::clickLeft(tribool down, bool previousState)
 	if(previousState && (!down))
 	{
 		int offset = LOCPLINT->castleInt? (-87) : 0;
+
 		auto recruitCb = [=](CreatureID id, int count)
 		{
 			LOCPLINT->cb->recruitCreatures(town, town->getUpperArmy(), id, count, level);
 		};
+
```
Comment (Nordsoft91) [inline client/windows/CCastleInterface.cpp:1065]:
```suggestion
```

File: client/windows/CCastleInterface.cpp [hunk 2]
```diff
@@ -1042,10 +1057,12 @@ void CCreaInfo::clickLeft(tribool down, bool previousState)
 	if(previousState && (!down))
 	{
 		int offset = LOCPLINT->castleInt? (-87) : 0;
+
```
Comment (Nordsoft91) [inline client/windows/CCastleInterface.cpp:1060]:
```suggestion
```

**Pull request merged**